src/Scene/AnimationPlayer.js

```diff
@@ -0,0 +1,231 @@
+ *       - when Animation is stopped
+ *       - when Animation is ending
+ */
+class AnimationPlayer {
+    constructor(dom) {
+        this.dom = dom;
+        this.id = null;
+        this.keyframe = 0;
+        this.animation = null;
+        this.resolve = null;
+        this.promise = null;
+        this.state = PLAYER_STATE.STOP;
+        this.callBackFrame = () => {};
+        this.waitTimer = null;
+    }
+
+    /**
+     * Gets the interval time.
+     *
+     * @return {Number}  The interval time is the time between consecutive frames
+     */
+    static get intervalTime() {
+        return INTERVAL_TIME;
+    }
+
+    isPlayed() {
+        return this.state === PLAYER_STATE.PLAY;
+    }
+
+    isStopped() {
+        return this.state === PLAYER_STATE.STOP;
+    }
+
+    isEnded() {
+        return this.state === PLAYER_STATE.END;
+    }
+
+    // Public functions
+
+    /**
+     * { Start animation }
+     *
+     * @param      {Animation} The animation to play
+     * @return     {Promise}  Promise is resolved when animation is stopped or finished
+     */
+    play(animation) {
+        this.animation = animation;
+        if (this.animation.animate) this.callBackFrame = () => this.animation.animate(this.keyframe);
+        setPlayerState(this, PLAYER_STATE.PLAY);
+        resetTimer(this);
+        this.id = setInterval(this.frame.bind(this), INTERVAL_TIME);
+        this.promise = new Promise(r => this.resolve = r);
+        return this.promise;
+    }
+
+    /**
+     * { The animation is played after a number of frames }
+     *
+     * @param      {Animation}  animation    The animation to play
+     * @param      {Number}  waitingTime  The waiting time before start animation (time in frame)
+     */
+    playLater(animation, waitingFrame) {
+        this.resolveWait = null;
+        const promise = new Promise(r => this.resolveWait = r);
+        const timew = Math.floor(INTERVAL_TIME * waitingFrame);
+        window.clearInterval(this.waitTimer);
+        this.waitTimer = window.setTimeout(() => { this.play(animation).then(() => this.resolveWait()); }, timew);
+        return promise;
+    }
+
+    /**
+     * { Stop current animation }
+     *
+     * @return  {Promise}  Promise is resolved when animation is stopped or finished
+     */
+    stop() {
+        setPlayerState(this, PLAYER_STATE.STOP);
+        finishAnimation(this);
+        this.dom.dispatchEvent(stopEvent);
+        // needed to return promise to wait sync
+        return Promise.resolve();
+    }
+
+    /**
+     * { Pause current animation }
+     */
+    pause() {
+        setPlayerState(this, PLAYER_STATE.PAUSE);
+        clearInterval(this.id);
+        this.id = undefined;
+    }
+
+    /**
+     * { this function is executed with each frame }
+     */
+    frame() {
+        if (this.keyframe < this.animation.duration) {
+            this.callBackFrame();
+            this.keyframe++;
+            this.dom.dispatchEvent(frameEvent);
```

Comment:
`callBackFrame` is function to animate object
`frameEvent` is used to resfresh display.